scripts/generate_black_spots_input.py

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env python
+
+"""Loads records and roads shapefile, outputs data needed for generating black spots"""
+import argparse
+import csv
+from dateutil import parser
+from dateutil.relativedelta import relativedelta
+import fiona
+from functools import partial
+import itertools
+import logging
+import os
+import pyproj
+import pytz
+import rtree
+from shapely.geometry import mapping, shape, LineString, Point
+from shapely.ops import transform, unary_union
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger()
+
+
+def read_roads(roads_shp):
+    """Reads shapefile and extracts roads and projection"""
+    shp_file = fiona.open(roads_shp)
+    return ([shape(road['geometry']) for road in shp_file], shp_file.crs)
+
+
+def get_intersections(roads):
+    """Calculates the intersection points of all roads"""
```

Comment:
It'd be good to include a :param: description for `roads` so it's clear what type of objects they are.